DemoApp/StreamChat/Components/DemoChatMessageContentView.swift

```diff
@@ -16,9 +26,22 @@ final class DemoChatMessageContentView: ChatMessageContentView {
 
         if let translations = content?.translations, let turkishTranslation = translations[.turkish] {
             textView?.text = turkishTranslation
-            if let timestampLabelText = timestampLabel?.text {
-                timestampLabel?.text = "\(timestampLabelText) - Translated to Turkish"
+            timestampLabel?.text?.append(" - Translated to Turkish")
+        }
+        
+        if content?.isPinned == true {
+            backgroundColor = UIColor(red: 0.984, green: 0.957, blue: 0.867, alpha: 1)
+            pinInfoLabel.text = "📌 Pinned"
+            if let pinDetails = content?.pinDetails {
+                let pinnedByName = content?.isSentByCurrentUser == true
+                    ? (content?.author.id == pinDetails.pinnedBy.id ? "You" : pinDetails.pinnedBy.name ?? pinDetails.pinnedBy.id)
+                    : pinDetails.pinnedBy.name ?? pinDetails.pinnedBy.id
+                pinInfoLabel.text?.append(" by \(pinnedByName)")
             }
+            bubbleThreadFootnoteContainer.insertArrangedSubview(pinInfoLabel, at: 0)
+        } else {
+            backgroundColor = nil
+            pinInfoLabel.removeFromSuperview()
```

Comment:
Re-layouting the cell when content changes badly affects message list performance. To avoid this, we recommend customers have a separate layout option if they need a custom subview. It affects the reuse identifier so that when you get a message, the cell suitable for this message (with all subviews in place) is returned. 

This way, in `layout` you'll check the option is present:

```swift
if options.contains(.pinnedInfo) {
 pinInfoLabel = UILabel()
 bubbleThreadFootnoteContainer.insertArrangedSubview(pinInfoLabel!, at: 0)
}
```
and in `updateContent` it'll receive the correct text:
```swift
pinInfoLabel?.text = content?.pinInformation
```

Reply:
Alright, good call 👍